backend-rust/src/scalar_types.rs

```diff
@@ -69,6 +73,30 @@ impl From<concordium_rust_sdk::common::types::Amount> for UnsignedLong {
     fn from(value: concordium_rust_sdk::common::types::Amount) -> Self { Self(value.micro_ccd()) }
 }
 
+impl TryFrom<f64> for UnsignedLong {
+    type Error = anyhow::Error;
+
+    fn try_from(value: f64) -> Result<Self, Self::Error> {
+        if value < 0.0 {
+            return Err(anyhow::anyhow!(
+                "Failed to convert f64 to `UnsignedLong` scalar type. Actual value is negative: \
+                 {:?}.",
+                value
+            ));
+        }
+
```

Comment:
I hope we can avoid having this conversion by using `u128` as suggested by @td202, since it loose precision, but in case there is really no other way then we should probably round before truncating the decimals from the casting.
```suggestion
    let value = value.round();
```